src/components/Navigation/Navigation.jsx

```diff
@@ -0,0 +1,141 @@
+import "./Navigation.css";
+import HomeLogoutIcon from "../../images/wLogout.svg";
+import SaveNewsLogoutIcon from "../../images/bLogout.svg";
+import HmenuIcon from "../../images/homeMenu.svg";
+import SmenuIcon from "../../images/savedMenu.svg";
+import { useContext, useEffect, useState } from "react";
+import { Link, useLocation } from "react-router-dom";
+import CurrentUserContext from "../../contexts/CurrentUserContext";
+import MobileMenu from "../MobileMenu/MobileMenu";
+
+function Nav({ handleSignInClick, isLoggedIn, onLogOut, activeModal }) {
+  const [activeItem, setActiveItem] = useState("home");
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const location = useLocation();
+
+  const isHomePage = location.pathname === "/";
+  const isSavedNewsPage = location.pathname === "/saved-news";
+  const currentUser = useContext(CurrentUserContext);
+
+  useEffect(() => {
+    if (isHomePage) {
+      setActiveItem("home");
+    } else if (isSavedNewsPage) {
+      setActiveItem("saved");
+    }
+  }, [location.pathname, isHomePage, isSavedNewsPage]);
+
+  const handleSignInAndCloseMenu = () => {
+    setIsMenuOpen(false);
+    handleSignInClick();
+  };
+
+  return (
+    <nav className="nav">
+      <div
+        className={`nav__container ${
+          activeItem === "saved" ? "nav__container--black-text" : ""
+        }`}
+      >
+        <ul className="nav__list">
+          <li
+            className={`nav__item nav__item--logo ${
+              activeModal ? "nav__hide-mobile" : ""
+            }`}
+          >
+            <Link to="/" className="nav__link">
+              NewsExplorer
+            </Link>
+          </li>
+
+          <li
+            className={`nav__item nav__item--home ${
+              isSavedNewsPage ? "nav__item--home-hover" : ""
+            }`}
+          >
+            <Link
+              to="/"
+              className={`nav__link ${
+                activeItem === "home" ? "nav__item--active" : ""
```

Comment:
`nav__item--active` is a **modifier** of `nav__item`, but you have `nav__link` for the links. Please, use `nav__link--active`

**Modifier** should be used only with its modifiable **basic** class: `block__element block__element_modifier-key_modifier-value`. Elements should get only **modifying styles** from modifier, and all other styles should be got from the basic class.


**Please, correct it everywhere in the code**